### PR TITLE
Expose additional booleans in sd_journal and improve error messages in journalctl

### PR DESCRIPTION
**Pull request by jsynacek**

### CODE REVIEW

File: src/journal/sd-journal.c [hunk 1]
```diff
@@ -2630,3 +2641,11 @@ _public_ int sd_journal_get_data_threshold(sd_journal *j, size_t *sz) {
         *sz = j->data_threshold;
         return 0;
 }
+
+_public_ bool sd_journal_has_runtime_files(sd_journal *j) {
```
Comment (zonque) [inline src/journal/sd-journal.c:2645]: `assert(j);`

File: src/journal/sd-journal.c [hunk 2]
```diff
@@ -2630,3 +2641,11 @@ _public_ int sd_journal_get_data_threshold(sd_journal *j, size_t *sz) {
...
+        return j->has_runtime_files;
+}
+
+_public_ bool sd_journal_has_persistent_files(sd_journal *j) {
+        return j->has_persistent_files;
```
Comment (zonque) [inline src/journal/sd-journal.c:2650]: `assert(j);`

**Comment by poettering** [comment]
Looks good otherwise.

**Comment by jsynacek** [comment]
v3: https://github.com/systemd/systemd/pull/2353

**Pull request closed without merging**